go/src/url_shortener/url_shortener.go

```diff
@@ -114,8 +123,52 @@ func (c *URLShortener) shortenHandler(w http.ResponseWriter, r *http.Request) {
 	c.incrementHandlerCounter(c.shortenRoute, true)
 }
 
+type redirectsJSON struct {
+	Success int `json:"success"`
+	Failed  int `json:"failed"`
+}
+
+type handlerJSON struct {
+	Name  string `json:"name"`
+	Count int    `json:"count"`
+}
+
+type serverStatsJSON struct {
+	TotalURL  int           `json:"total_url"`
+	Redirects redirectsJSON `json:"redirects"`
+	Handlers  []handlerJSON `json:"handlers"`
+}
+
+type statsJSON struct {
+	ServerStats serverStatsJSON `json:"server_stats"`
+}
+
 func (c *URLShortener) statisticsHandler(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "%s", c.getStatistics())
+	url := r.URL
+	query := url.Query()
```

Comment:
You can use query.Get("format') which returns a string here so you don't have to deal with a []string and can remove the loop below

Reply:
What happens if 2 `format` parameters are passed to the query and the `json` is the second?
Something like: `http://localhost:9090/statistics?format=foo&format=json`.
Or do I need to do not care?

Reply:
True, with my proposition only the first would be taken into account. But I think we don't care since the specification does't indicate that supplying twice the same parameter is supported, so it's undefined behaviour :-)